compiler/src/dmd/hdrgen.d

```diff
@@ -2724,6 +2724,12 @@ private void expressionPrettyPrint(Expression e, ref OutBuffer buf, ref HdrGenSt
 
     void visitComma(CommaExp e)
     {
+        if (e.originalExp !is null)
+        {
+            e.originalExp.expressionPrettyPrint(buf, hgs);
```

Comment:
Can you add a test case for this, by adding an error related to `x ^^ 2` to a file in fail_compilation that prints an error message with it? You can use an existing test file as a template for getting a TEST_OUTPUT block, and run `rdmd run.d your_test_name.d AUTO_UPDATE=1` to automatically update the TEST_OUTPUT block.

Reply:
Addressed in the last commit, https://github.com/dlang/dmd/pull/21082/commits/72754cdebc284e7d87d8dd693d5b2d41a676fdf7.